Gems/ROS2/Code/Source/Manipulation/JointsPositionsEditorComponent.cpp

```diff
@@ -76,14 +76,25 @@ namespace ROS2
 
     AZ::Crc32 JointsPositionsEditorComponent::FindAllJoints()
     {
+        m_jointNames.clear();
         AZStd::function<void(const AZ::Entity* entity)> getAllJointsHierarchy = [&](const AZ::Entity* entity)
         {
             auto* frameEditorComponent =
                 azrtti_cast<ROS2::ROS2FrameEditorComponent*>(Utils::GetGameOrEditorComponent<ROS2::ROS2FrameEditorComponent>(entity));
             AZ_Assert(frameEditorComponent, "ROS2FrameEditorComponent does not exist!");
 
+            const bool hasPrismaticJoints = Utils::CheckIfEntityHasComponentOfType(
```

Comment:
It is unclear why classical joints are handled here and only articulations are handled in Reload. Looks incorrect.